### ISSUE TITLE
Can we add link anchors to headings in MD pages?

### ISSUE THREAD

**Issue by jgeewax** [GoogleCloudPlatform/gcloud-common#137]
I want to link to a specific heading on the overview page on gcloud-ruby, and can't unless I open up the inspector and find the ID for the heading and then do it manually.

Can we make headings into hoverable links so I can right click and do "copy link location" to share a specific section?

Ie, https://quartzmo.github.io/gcloud-ruby/#/docs/master/gcloud/datastore#paginating-records (I had to create that myself)

/cc @quartzmo @callmehiphop

**Comment by stephenplusplus** [comment]
Duplicate: https://github.com/GoogleCloudPlatform/gcloud-common/issues/136 -- TLDR, we may have an ingenious fix. We just need a volunteer to try it.

**Comment by callmehiphop** [comment]
@stephenplusplus I think the functionality between that issue and this request are slightly different. While #136 would allow us to use the `#` for deep linking, I believe this issue is asking for heading tags in markdown files to generate a deep link url when clicked (regardless of whether we use the `#` or a query string).

**Comment by stephenplusplus** [comment]
Okay, pretending this never happened.

**Comment by callmehiphop** [comment]
Looks like this isn't an option of showdown, not from what I can tell at least. @stephenplusplus did we have deep linking capabilities into the markdown guides on the old site? Either way, I think we could just find/write an extension to do it.

**Comment by jgeewax** [comment]
Bump :)

Trying to link again to specific sections (and examples) and I can't. Which makes me sad.

![](http://www.mememaker.net/static/images/memes/3675821.jpg)

**Comment by jgeewax** [comment]
@callmehiphop : https://github.com/showdownjs/showdown/compare/master...jgeewax:patch-1

Thoughts?

**Comment by callmehiphop** [comment]
@jgeewax I think the issue here is that the `#` is already spoken for by Angular.

We need to either:

1. Resolve #136 
2. Add a custom extension to add deeplinking

**Comment by daspecster** [comment]
@callmehiphop what do you think of that `404.html` idea?

**Comment by callmehiphop** [comment]
@daspecster I haven't looked into it very deeply (yet), but it seems pretty clever and the rest of the group seems to be in favor of it.

**Comment by jgeewax** [comment]
As much as a I love the ingenuity.... it feels like one of those things that will bite us in the ass later, and then we'll be too far committed to it to fix things properly. 

Can we explore making the header ID a sub-resource? ie,

```html
<h3 id="#/docs/v0.11.0/gcloud/storage/file/upload/example-1">
  <a href="#/docs/v0.11.0/gcloud/storage/file/upload/example-1">Upload a file</a>
</h3>
```

**Comment by quartzmo** [comment]
I think this can be closed as a duplicate of #40. Any objections?
